app/models/runtime/user.rb

```diff
@@ -163,48 +163,62 @@ def remove_spaces(space)
     end
 
     def membership_spaces
-      Space.join(:spaces_developers, space_id: :id, user_id: id).select(:spaces__id).
-        union(
-          Space.join(:spaces_auditors, space_id: :id, user_id: id).select(:spaces__id)
-        ).
-        union(
-          Space.join(:spaces_managers, space_id: :id, user_id: id).select(:spaces__id)
-        )
+      Space.where(id: membership_space_ids).select(:id)
```

Comment:
Also returns the space ID when the user is space_supporter. Dont know if this is right since the previous logic does not include space_supporters.

Reply:
Ok so on closer inspection it turns out that the user model's `membership_spaces` is called by (and only by) the service_plan model ([here](https://github.com/cloudfoundry/cloud_controller_ng/blob/8173f4677b350b79dc5c985a3005119b9644919a/app/models/services/service_plan.rb#L129-L131)). From what I can tell running through the unit tests with the debugger, only v2 code depends on it (in particular, via some indirection, the [v2 service instances controller](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/app/controllers/services/service_instances_controller.rb)). I've verified that service plans are visible to space supporters using the v3 API, and a complete run of v3 request tests with the debugger didn't result in `membership_spaces` being invoked.

I guess there are two options:
1. Since I'm here anyway and it's quick, change this into a union of the three space role tables that were previously being joined, perhaps with a comment pointing out that a refactor is needed if any future v3 use is made of it.
2. Just revert it back to the way it was before I turned up, since it's a v2-only thing and we're not really aiming to optimize that part of the codebase.

Any preference @FloThinksPi?

Reply:
Puh whatever seems less work for you. Both are fine for me. Would probably go for a revert of this function since then the change is a no brainer and we know it works :)

Reply:
Cool, yeah I've done that. Rebased and force-pushed.